language_service/src/hover.rs

```diff
@@ -307,6 +358,16 @@ fn is_param(param_pats: &[&ast::Pat], node_id: ast::NodeId) -> bool {
     param_pats.iter().any(|pat| find_in_pat(pat, node_id))
 }
 
+fn is_struct(ty_def: &ast::TyDef) -> bool {
+    match ty_def.kind.as_ref() {
+        ast::TyDefKind::Paren(inner) => is_struct(inner),
+        ast::TyDefKind::Tuple(fields) => fields
+            .iter()
+            .all(|field| matches!(field.kind.as_ref(), ast::TyDefKind::Field(Some(_), _))),
+        ast::TyDefKind::Err | ast::TyDefKind::Field(..) => false,
+    }
+}
```

Comment:
I kind of expected this to be a method of `ast::TyDef`, just like `ty::Udt::is_struct()` is a method too. Would that make sense?

Reply:
good idea, done